test/test_transforms_v2_refactored.py

```diff
@@ -2185,7 +2161,9 @@ def test_unsupported_types(self, dispatcher, make_input):
 
 class TestRegisterKernel:
     @pytest.mark.parametrize("dispatcher", (F.resize, "resize"))
-    def test_register_kernel(self, dispatcher):
+    def test_register_kernel(self, mocker, dispatcher):
+        mocker.patch.dict(_KERNEL_REGISTRY, values={F.resize: _KERNEL_REGISTRY[F.resize]}, clear=True)
```

Comment:
@NicolasHug I missed this in my review of #7796. I'm a little anxious that we actually modified the registry here. I've mocked it out.

Reply:
Do we care? It's registering a value for `CustomDatapoint` that only exists in the scope of this test. It's never going to affect other tests in any way. Even registering another class named `CustomDatapoint` in another test won't be an issue (in fact this is what we do already in this test as it is parametrized, and if there was a conflict issue we would have had the "can't register a kernel twice" error).

If you'd really prefer cleaning up, I'd prefer to `del _KERNEL_REGISTRY[MyDispatcher]` at the end of this test, or have a `setup/teardown` autouse fixture for these tests, instead of using the `mocker` logic which is always acts in mysterious ways.